stock_picking_report_internal_delivery_address/models/stock_picking.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _is_to_external_location(self):
```

Comment:
This method may be used or other usages, so changing it only for getting an address shown seems a bad technique for me. You should instead override `should_print_delivery_address`:

https://github.com/OCA/OCB/blob/76076388cbb4d7e9a5f95b760ae4a0f3786d2fb3/addons/stock/models/stock_picking.py#L1161